app/asset-manifest.js

```diff
@@ -1,13 +1,22 @@
 import environment from './config/environment';
 
-const metaName = environment.modulePrefix + '/asset-manifest';
+const modulePrefix = environment.modulePrefix;
+const metaName = `${modulePrefix}/asset-manifest`;
+const nodeName = `${modulePrefix}/node-asset-manifest`;
+
 let config = {};
 
 try {
-  const rawConfig = document.querySelector('meta[name="' + metaName + '"]').getAttribute('content');
-  config = JSON.parse(unescape(rawConfig));
+  // If we have a Node version of the asset manifest, use that for FastBoot and
+  // similar environments.
+  if (require.has(nodeName)) {
```

Comment:
This should shadow the native `require` which will make it work in `node`.

We intentionally ship this separately inside of the meta to allow it to change independently from the JS code. I believe that the primary path should be the attempt from the meta and the secondary be the attempt from the JS code.